lib/shoryuken/queue.rb

```diff
@@ -41,23 +38,55 @@ def receive_messages(options)
         map { |m| Message.new(client, self, m) }
     end
 
+    # Returns whether this queue is a FIFO queue or not.
+    # @return [TrueClass, FalseClass]
+    def is_fifo?
+      queue_attributes.attributes[FIFO_ATTRIBUTE] == 'true'
+    end
+
+    # Returns whether this queue has content based deduplication enabled or not.
+    # @return [TrueClass, FalseClass]
+    def has_content_deduplication?
+      queue_attributes.attributes[CONTENT_DEDUP_ATTRIBUTE] == 'true'
+    end
+
     private
 
+    FIFO_ATTRIBUTE = 'FifoQueue'
```

Comment:
Wow. I'm surprised that SQS would have a queue attribute that would generate an error if the value was false. I've opened an issue with their SDK. I'll amend the code as suggested.